base/arraymath.jl

```diff
@@ -26,9 +26,100 @@ julia> A
 """
 conj!(A::AbstractArray{<:Number}) = (@inbounds broadcast!(conj, A, A); A)
 
-for f in (:-, :conj, :real, :imag)
-    @eval ($f)(A::AbstractArray) = broadcast_preserving_zero_d($f, A)
-end
+"""
+    conj(A::AbstractArray)
+
+Return an array containing the complex conjugate of each entry in array `A`.
+
+Equivalent to `conj.(A)`, except when `A` is a 0-dimensional result,
+in which case a 0-dimensional container is returned (rather than a scalar).
+
+# Examples
+```jldoctest
+julia> A = [1+im 2-im; 2+2im 3+im]
+2×2 Matrix{Complex{Int64}}:
+ 1+1im  2-1im
+ 2+2im  3+1im
+
+julia> conj(A)
+2×2 Matrix{Complex{Int64}}:
+ 1-1im  2+1im
+ 2-2im  3-1im
```

Comment:
Could we make the examples more compact without loss? E.g.
```suggestion
julia> conj([1 2im; 3 4im])
2×2 Matrix{Complex{Int64}}:
 1+0im  0-2im
 3+0im  0-4im
```

Reply:
I just copied the existing example for `conj!`. I'd say it makes sense to print `A` as it makes it very easy to compare the two arrays.

Reply:
Fair point. I do wonder if they could all be made less of a visual puzzle, `2-im` vs `2+2im` etc. 

Do they need to be matrices? Something like `conj([1, 2im, 3 + 4im])` would be easy to compare.

Reply:
OK let's use a vector.